perun-core/src/main/java/cz/metacentrum/perun/core/impl/ServicesManagerImpl.java

```diff
@@ -899,4 +899,16 @@ public int getDestinationsCount(PerunSession sess) {
 		}
 	}
 
+	@Override
+	public void deleteDestination(PerunSession sess, Destination destination) throws DestinationAlreadyRemovedException, RelationExistsException {
+		try {
+			if (0 == jdbc.update("delete from destinations where id = ?", destination.getId())) {
+				throw new DestinationAlreadyRemovedException("Destination: " + destination);
+			}
+		} catch (DataIntegrityViolationException ex) {
+			throw new RelationExistsException("Destination: " + destination + " has existing relations.");
```

Comment:
Please pass also original exception into the `RelationExistsException`.
```java
throw new RelationExistsException("Destination: " + destination + " has existing relations.", ex);
```

Reply:
Done